js/src/framework.ts

```diff
@@ -43,25 +44,46 @@ export type EvalScorer<Input, Output> =
   | ((args: EvalScorerArgs<Input, Output>) => Score)
   | ((args: EvalScorerArgs<Input, Output>) => Promise<Score>);
 
+export type EvalMetadata = {
+  experimentName?: string;
+};
+
+export function evalMetadataToInitOptions(
+  metadata: EvalMetadata | undefined
+): InitOptions {
+  return { experiment: metadata?.experimentName };
+}
+
 /**
  * An evaluator is a collection of functions that can be used to evaluate a model.
  * It consists of:
  * - `data`, a function that returns a list of inputs, expected outputs, and metadata
  * - `task`, a function that takes an input and returns an output
  * - `scores`, a set of functions that take an input, output, and expected value and return a score
+ * - `metadata`, optional additional metadata for the eval definition, such as experiment name.
  */
 export interface Evaluator<Input, Output> {
   data: EvalData<Input, Output>;
   task: EvalTask<Input, Output>;
   scores: EvalScorer<Input, Output>[];
+  metadata?: EvalMetadata;
+}
+
+function makeEvalName(projectName: string, metadata: EvalMetadata | undefined) {
+  let out = projectName;
+  if (metadata?.experimentName) {
+    out += ` [experimentName=${metadata.experimentName}]`;
+  }
+  return out;
 }
 
 export type EvaluatorDef<Input, Output> = {
-  name: string;
+  evalName: string;
+  projectName: string;
```

Comment:
nit: put `projectName` first since it's the first/primary name the user specifies